### PR TITLE
PIM-9084: Block the export of attributes with specific locale wich isn't the used locale in the export profile

### PR DESCRIPTION
**Pull request by anaisbaunelemaire**
**Description (for Contributor and Core Developer)**

**Issue**: The locale-specific attributes used to be exported even if the export profile was not set on their locale.

**Solution**: I added a way to check if the locale specific of an attribute is in the export profile available locales.
So, now, the attribute is exported only if its specific locale is set on the export profile.

**Definition Of Done (for Core Developer only)**

| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | 
| Added legacy Behats               | 
| Added acceptance tests            | 
| Added integration tests           | 
| Changelog updated                 | OK
| Review and 2 GTM                  | 
| Micro Demo to the PO (Story only) | 
 Migration script                  | -
| Tech Doc                          | -

`Todo`: Pending / Work in progress
`OK`: Done / Validated
`-`: Not needed

### CODE REVIEW

File: src/Akeneo/Pim/Enrichment/Component/Product/Connector/Processor/Normalization/ProductProcessor.php [hunk 1]
```diff
@@ -149,9 +149,15 @@ protected function filterValues(array $values, array $attributesToFilter)
...
-                $valuesToExport[$code] = $value;
+        foreach ($values as $attributeCode => $value) {
+            $attribute = $this->attributeRepository->findOneByIdentifier($attributeCode);
+            if ($attribute->isLocaleSpecific()) {
+                $availableCodes = $attribute->getLocaleSpecificCodes();
+                if (in_array($this->currentLocale, $availableCodes)) {
+                    if (isset($attributesToFilter[$attributeCode])) {
+                        $valuesToExport[$attributeCode] = $value;
+                    }
+                }
```
Comment (pierrejolly) [inline src/Akeneo/Pim/Enrichment/Component/Product/Connector/Processor/Normalization/ProductProcessor.php:160]: Maybe you can use something like this ?

```suggestion
            if ($attribute->isLocaleSpecific()
                && in_array($this->currentLocale, $attribute->getLocaleSpecificCodes())
                && isset($attributesToFilter[$attributeCode])
            ) {
                $valuesToExport[$attributeCode] = $value;
            }
```

File: src/Akeneo/Pim/Enrichment/Component/Product/Connector/Processor/Normalization/ProductProcessor.php [hunk 2]
```diff
@@ -151,7 +151,12 @@ protected function filterValues(array $values, array $attributesToFilter)
         $attributesToFilter = array_flip($attributesToFilter);
         foreach ($values as $code => $value) {
             if (isset($attributesToFilter[$code])) {
-                $valuesToExport[$code] = $value;
+                $attribute = $this->attributeRepository->findOneByIdentifier($code);
+                $parameters = $this->stepExecution->getJobParameters();
+                $currentLocale = $parameters->get('filters')['structure']['locales'];
```
Comment (jmleroux) [inline src/Akeneo/Pim/Enrichment/Component/Product/Connector/Processor/Normalization/ProductProcessor.php:156]:
```suggestion
                $currentLocale = $this->stepExecution->getJobParameters()->get('filters')['structure']['locales'];
```

**Comment by mmetayer** [comment]
Please add some tests :wink:

**Comment by pierrejolly** [comment]
GTM after you add tests ;)

File: src/Akeneo/Pim/Enrichment/Component/Product/Connector/Processor/Normalization/ProductProcessor.php (hunk 2 above)
Comment (anaisbaunelemaire) [reply src/Akeneo/Pim/Enrichment/Component/Product/Connector/Processor/Normalization/ProductProcessor.php:156]: I hesitated... Ok, I'll update it !

File: CHANGELOG-3.2.md [hunk 3]
```diff
@@ -1,5 +1,9 @@
 # 3.2.x
 
+## Bug fixes
+
+- PIM-9084: Block the export of attributes with specific locale wich isn't the used locale in the export profile
```
Comment (mmetayer) [inline CHANGELOG-3.2.md:5]:
```suggestion
- PIM-9084: Filter locale specific attributes in exports when the value's locale and the export profile's locale are different
```

File: src/Akeneo/Pim/Enrichment/Component/Product/Connector/Processor/Normalization/ProductProcessor.php [hunk 4]
```diff
@@ -158,6 +160,20 @@ protected function filterValues(array $values, array $attributesToFilter)
         return $valuesToExport;
     }
 
+    protected function filterLocaleSpecificAttribute(array $values): array
```
Comment (mmetayer) [inline src/Akeneo/Pim/Enrichment/Component/Product/Connector/Processor/Normalization/ProductProcessor.php:163]:
```suggestion
    protected function filterLocaleSpecificAttributes(array $values): array
```

File: tests/back/Pim/Enrichment/Specification/Component/Product/Connector/Processor/Normalization/ProductProcessorSpec.php [hunk 5]
```diff
@@ -3,6 +3,8 @@
 namespace Specification\Akeneo\Pim\Enrichment\Component\Product\Connector\Processor\Normalization;
 
 use Akeneo\Pim\Enrichment\Component\Product\Connector\Processor\Normalization\ProductProcessor;
+use Akeneo\Pim\Structure\Bundle\Doctrine\ORM\Repository\AttributeRepository;
```
Comment (mmetayer) [inline tests/back/Pim/Enrichment/Specification/Component/Product/Connector/Processor/Normalization/ProductProcessorSpec.php:6]: not needed, use the already present AttributeRepositoryInterface

File: tests/back/Pim/Enrichment/Specification/Component/Product/Connector/Processor/Normalization/ProductProcessorSpec.php [hunk 6]
```diff
@@ -200,8 +211,14 @@ function it_throws_an_exception_if_media_of_product_is_not_found(
         JobExecution $jobExecution,
         JobInstance $jobInstance,
         WriteValueCollection $valuesCollection,
-        ExecutionContext $executionContext
+        ExecutionContext $executionContext,
+        AttributeRepository $attributeRepository,
```
Comment (mmetayer) [inline tests/back/Pim/Enrichment/Specification/Component/Product/Connector/Processor/Normalization/ProductProcessorSpec.php:215]:
```suggestion
        AttributeRepositoryInterface $attributeRepository,
```

Comment (anaisbaunelemaire) [reply tests/back/Pim/Enrichment/Specification/Component/Product/Connector/Processor/Normalization/ProductProcessorSpec.php]: The test doesn't work if I use the existed AttributeRepositoryInterface...

**Pull request merged**